automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/methodconfigs/WorkspaceMethodConfigDetailsPage.scala

```diff
@@ -33,35 +34,49 @@ class WorkspaceMethodConfigDetailsPage(namespace: String, name: String, methodCo
   private val modalConfirmDeleteButton = Button("modal-confirm-delete-button")
   private val snapshotRedactedLabel = Label("snapshot-redacted-title")
 
-  def clickLaunchAnalysis(): Unit = {
+  def clickLaunchAnalysis[T <: FireCloudView](page: T): T = {
     openLaunchAnalysisModalButton.doClick()
     Try(
-      // after click, expect to find either a Message or Analysis Modal. If not found, retry click
-      await condition (find(CssSelectorQuery(".broadinstitute-modal-open")).nonEmpty, 5)
+      page.awaitReady() // check if click happened
```

Comment:
wait for expected page ready in 30 sec. if errored, try click Launch Analysis button again